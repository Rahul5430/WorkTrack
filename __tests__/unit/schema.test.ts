describe('Database Schemas', () => {
	it('imports and snapshots all schemas', () => {
		// Import all schema files
		const appSchema = require('../../src/db/watermelon/schema');
		const worktrackSchema = require('../../src/db/watermelon/worktrack/schema');

		// Test that schemas are defined
		expect(appSchema.default).toBeDefined();
		expect(worktrackSchema.workTrackSchema).toBeDefined();

		// Snapshot the schema structures
		expect(appSchema.default).toMatchSnapshot('app-schema');
		expect(worktrackSchema.workTrackSchema).toMatchSnapshot(
			'worktrack-schema'
		);
	});

	it('verifies schema table names', () => {
		const appSchema = require('../../src/db/watermelon/schema');

		// Test that the app schema contains expected table names
		expect(appSchema.default.tables).toBeDefined();
		expect(Array.isArray(appSchema.default.tables)).toBe(true);
		expect(appSchema.default.tables.length).toBeGreaterThan(0);

		// Check that tables have the expected structure
		const tableNames = appSchema.default.tables.map(
			(table: { name: string }) => table.name
		);
		expect(tableNames).toContain('work_tracks');
		expect(tableNames).toContain('trackers');
		expect(tableNames).toContain('shared_trackers');
	});

	it('verifies schema version', () => {
		const appSchema = require('../../src/db/watermelon/schema');

		// Test that schema has a version
		expect(appSchema.default.version).toBeDefined();
		expect(typeof appSchema.default.version).toBe('number');
		expect(appSchema.default.version).toBeGreaterThan(0);
	});
});
