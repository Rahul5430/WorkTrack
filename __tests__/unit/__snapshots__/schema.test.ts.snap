// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Database Schemas imports and snapshots all schemas: app-schema 1`] = `
{
  "tables": [
    {
      "columns": [
        {
          "isIndexed": true,
          "name": "date",
          "type": "string",
        },
        {
          "name": "status",
          "type": "string",
        },
        {
          "name": "created_at",
          "type": "number",
        },
        {
          "name": "last_modified",
          "type": "number",
        },
        {
          "name": "is_advisory",
          "type": "boolean",
        },
        {
          "isIndexed": true,
          "name": "tracker_id",
          "type": "string",
        },
        {
          "name": "needs_sync",
          "type": "boolean",
        },
        {
          "isOptional": true,
          "name": "sync_error",
          "type": "string",
        },
        {
          "isOptional": true,
          "name": "retry_count",
          "type": "number",
        },
      ],
      "name": "work_tracks",
    },
    {
      "columns": [
        {
          "name": "owner_id",
          "type": "string",
        },
        {
          "name": "name",
          "type": "string",
        },
        {
          "name": "color",
          "type": "string",
        },
        {
          "name": "created_at",
          "type": "number",
        },
        {
          "name": "is_default",
          "type": "boolean",
        },
        {
          "name": "tracker_type",
          "type": "string",
        },
      ],
      "name": "trackers",
    },
    {
      "columns": [
        {
          "name": "tracker_id",
          "type": "string",
        },
        {
          "name": "shared_with",
          "type": "string",
        },
        {
          "name": "permission",
          "type": "string",
        },
        {
          "name": "created_at",
          "type": "number",
        },
      ],
      "name": "shared_trackers",
    },
  ],
  "version": 4,
}
`;

exports[`Database Schemas imports and snapshots all schemas: worktrack-schema 1`] = `
{
  "columns": [
    {
      "isIndexed": true,
      "name": "date",
      "type": "string",
    },
    {
      "name": "status",
      "type": "string",
    },
    {
      "name": "created_at",
      "type": "number",
    },
    {
      "name": "last_modified",
      "type": "number",
    },
    {
      "name": "is_advisory",
      "type": "boolean",
    },
    {
      "isIndexed": true,
      "name": "tracker_id",
      "type": "string",
    },
    {
      "name": "needs_sync",
      "type": "boolean",
    },
    {
      "isOptional": true,
      "name": "sync_error",
      "type": "string",
    },
    {
      "isOptional": true,
      "name": "retry_count",
      "type": "number",
    },
  ],
  "name": "work_tracks",
}
`;
